src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/validators/OcrDataValidatorTest.java

```diff
@@ -0,0 +1,229 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.validators;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrDataField;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.OcrValidationResult;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.ValidationStatus;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.services.OcrDataValidator;
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.FormType.CONTACT;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.FormType.PERSONAL;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.ADDRESS_LINE_1;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.CONTACT_NUMBER;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.COUNTRY;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.DATE_OF_BIRTH;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.EMAIL;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.FIRST_NAME;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.LAST_NAME;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.POST_CODE;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.ValidationStatus.ERRORS;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.ValidationStatus.SUCCESS;
+
+class OcrDataValidatorTest {
+
+    private final OcrDataValidator validator = new OcrDataValidator();
+
+    @Test
+    void should_return_errors_when_mandatory_fields_are_missing() {
+        // given
+        List<OcrDataField> ocrDataFields = singletonList(new OcrDataField(FIRST_NAME, "test"));
+
+        // when
+        OcrValidationResult result = validator.validate(PERSONAL, ocrDataFields);
+
+        // then
+        assertThat(result).isNotNull();
+        assertThat(result)
+            .extracting("errors", "warnings", "status")
+            .containsExactly(
+                singletonList(String.format("%s is missing", LAST_NAME)),
```

Comment:
Is there a reason we are doing the check like this ?

Reply:
checking all fields in the result in a single assert. happy to change it